### ISSUE TITLE
pulumi login should not be a global user-level configuration

### ISSUE THREAD

**Issue by ringods** [pulumi/pulumi#3747]
Our company helps other technology companies with improving their cloud-native architecture & delivery automation. We use Pulumi ourselves for our own infrastructure, but this is not much.

When we use Pulumi for customers, I think it is clear that each of our customers use a different backend for their set of projects. At the moment, I have to login/logout at least once a day to switch between the backend of our own projects and the backend of the customer projects. I have also made a few errors already where I forget this and the state of one customer ends up in the backend of another, or in ours.

This is annoying. Please search for a way that the `pulumi` CLI could pick up the backend configuration & credentials from a local file or environment variables. 

Personally, I would like to have environment variables as the means to pick up the Pulumi login configuration. I'm a user of [direnv](https://direnv.net/) to have specific environment variables loaded when i `cd` into a certain folder and these are unset when I `cd` out of it. Each shell can this way have different settings.

**Comment by lukehoban** [comment]
Yes - this idea that the backend to use should really be information tied to an individual stack has come up several times.  

The one case it doesn’t directly handle is creation of a new stack - though it could be that passing the backend as a parameter to that command is the right approach. 

We have `—secrets-provider` which is a similar idea, and behaves like that (stored with the stack, passed initially to stack knit). 

Your suggestion of an environment variable is another option to consider.

**Comment by ringods** [comment]
@lukehoban in this issue I'm not really asking to tie a backend to a project or a stack. 

For me, being able to "link" the customer backend to the set of projects from that same customer, and switch without too much hassle to our own setup. I can live with a single backend for a set of projects, but per customer.

**Comment by ringods** [comment]
@lukehoban will this be tackled shortly? 

The user-level login scope prevents me of provisioning infrastructure for different customers at once. And since infra provisioning usually takes a while, this is a major blocker to my productivity. 😢
